### ISSUE TITLE
Broken code with html-self-closing rule

### ISSUE THREAD

**Issue by scriptPilot** [vuejs/eslint-plugin-vue#876]
**Tell us about your environment**	
* **ESLint version:** 5.16.0
* **eslint-plugin-vue version:** 5.2.2	
* **Node version:**	 11.6.0

**Please show your full configuration:**

```json
{
  "extends": [
    "airbnb-base",
    "plugin:vue/recommended"
  ]
}
```

**What did you do?**	

Created the following vue component / fixed with ESLint according config:

```vue
<template>
  <f7-app :params="f7params">
    <f7-statusbar />
    <f7-view main />
  </f7-app>
</template>
```

**What did you expect to happen?**
https://framework7.io/ app component is bundled nicely with https://parceljs.org/.

**What actually happened?**
Resulting code is broken. Bundling works nicely when I add the following rule:

```json
"rules": {
    "vue/html-self-closing": [
      "error",
      {
        "html": {
          "component": "never"
        }
      }
    ]
  }
```

Resulting in the following vue component:

```vue
<template>
  <f7-app :params="f7params">
    <f7-statusbar></f7-statusbar>
    <f7-view main></f7-view>
  </f7-app>
</template>
```

**Comment by ota-meshi** [comment]
Thank you for this issue.

I don't know because I've never used parcel, but both code should build successfully.
If that doesn't work, change the rule settings and use it.
